tasks/release/ReleaseVersionController.php

```diff
@@ -36,15 +36,50 @@ function assignVersionToCompletedTickets($version = null) {
     return [$version, join("\n", $output)];
   }
 
-  private function getDoneIssuesWithoutVersion() {
-    $jql = "project = $this->project AND status = Done AND fixVersion = EMPTY AND updated >= -52w";
-    $result = $this->jira->search($jql, ['key']);
-    return array_map(function ($issue) {
-      return [
-        'id' => $issue['id'],
-        'key' => $issue['key'],
-      ];
-    }, $result['issues']);
+  function determineNextVersion() {
+    $last_version = $this->jira->getLastReleasedVersion();
+    $parsed_last_version = VersionHelper::parseVersion($last_version['name']);
+
+    $version_to_increment = VersionHelper::PATCH;
+
+    if ($this->project === JiraController::PROJECT_MAILPOET) {
+      $free_increment = $this->checkProjectVersionIncrement(JiraController::PROJECT_MAILPOET);
+      $premium_increment = $this->checkProjectVersionIncrement(JiraController::PROJECT_PREMIUM);
+
+      if (in_array(VersionHelper::MINOR, [$free_increment, $premium_increment])) {
+        $version_to_increment = VersionHelper::MINOR;
+      }
+    }
+
+    $parsed_last_version[$version_to_increment]++;
+    $next_version = VersionHelper::buildVersion($parsed_last_version);
+    return $next_version;
+  }
+
+  private function checkProjectVersionIncrement($project) {
+    $this->jira->setProject($project);
+    $issues = $this->getUnreleasedDoneIssues();
+    $this->jira->setProject(JiraController::PROJECT_MAILPOET); // restore project
+
+    $version_to_increment = VersionHelper::PATCH;
+    $field_id = JiraController::VERSION_INCREMENT_FIELD_ID;
+
+    foreach ($issues as $issue) {
+      if (!empty($issue['fields'][$field_id]['value'])
+        && $issue['fields'][$field_id]['value'] === VersionHelper::MINOR
```

Comment:
Maybe a nicer coding style would be:
```
if (
  !empty(...)
  && $issue[...]
)
```
Not important though.

Reply:
I think the current one better adheres to PSR-2: 
> There MUST NOT be a space after the opening parenthesis

https://www.php-fig.org/psr/psr-2/#5-control-structures

Reply:
I think that's just for single-line conditions. They don't really specify multiline, however all other structures (function calls, etc.) specify using the style I suggested. Not important so let's leave it as is.